numpy/core/setup_common.py

```diff
@@ -124,9 +124,8 @@ def set_sig(sig):
         "floor", "ceil", "sqrt", "log10", "log", "exp", "asin",
         "acos", "atan", "fmod", 'modf', 'frexp', 'ldexp']
 
-# Standard functions which may not be available and for which we have a
-# replacement implementation. Note that some of these are C99 functions.
-OPTIONAL_STDFUNCS = ["expm1", "log1p", "acosh", "asinh", "atanh",
+# Some of these are C99 functions.
+MANDATORY_FUNCS += ["expm1", "log1p", "acosh", "asinh", "atanh",
         "rint", "trunc", "exp2", "log2", "hypot", "atan2", "pow",
         "copysign", "nextafter", "strtoll", "strtoull", "cbrt"]
 
```

Comment:
`hypot` and `atan2` are problematic on windows, see npy_config.h for the logic to disable the built-in functions and use the alternatives I removed. I guess I should move them to `OPTIONAL` and restore the code in `npy_math_internal.h.src`.

Reply:
`hypot` yes indeed. `atan2` is only undef'd for `MSC_VER < 1900`, so that is no longer relevant. You could clean it up in `npy_config.h` instead.

Reply:
> `atan2` is only undef'd for MSC_VER < 1900

It is also undef'd for `__MINGW32_VERSION`, which we do not test for. `MSC_VER 1900` corresponds to visual studio 2015. Officially its [exended end date](https://docs.microsoft.com/en-us/lifecycle/products/visual-studio-2015) is 2025. Do we have a clear policy for dropping older compilers?

Reply:
On the scipy-side, the minimum requiremen has moved to vs2019 already, because even vs2019 is already [EOL](https://docs.microsoft.com/en-us/lifecycle/products/visual-studio-2017). The Extended date is paid-only support AFAIK, and since vs2019 & vs2022 are ABI-compatible back to vs2015, I don't personally see a reason to keep supporting such old compilers.